backend/src/stwo/prover.rs

```diff
@@ -18,15 +40,102 @@ impl<F: FieldElement> StwoProver<F> {
     #[allow(dead_code)]
     #[allow(unused_variables)]
     pub fn new(
-        _analyzed: Arc<Analyzed<F>>,
+        analyzed: Arc<Analyzed<F>>,
         _fixed: Arc<Vec<(String, Vec<F>)>>,
-        setup: Option<&mut dyn io::Read>,
+        //  setup: Option<&mut dyn io::Read>,
     ) -> Result<Self, io::Error> {
         Ok(Self {
-            _analyzed,
+            analyzed,
             _fixed,
             _proving_key: None,
             _verifying_key: None,
         })
     }
+    pub fn prove(&self, witness: &[(String, Vec<F>)], witgen_callback: WitgenCallback<F>) {
+        let config = PcsConfig {
+            pow_bits: 16,                          // Any value you want to set for pow_bits
+            fri_config: FriConfig::new(0, 1, 100), // Using different numbers for FriConfig
+        };
+
+        //Trace
+        let circuit = PowdrCircuit::new(self.analyzed.clone())
+            .with_witgen_callback(witgen_callback.clone())
+            .with_witness(witness)
+            .generate_stwo_circuit_trace();
+       
+
+        let circuitEval = PowdrCircuit::new(self.analyzed.clone())
+            .with_witgen_callback(witgen_callback.clone())
+            .with_witness(witness);
```

Comment:
Ideally we'd create the circuit a single time, and then do everything based on that one instantiation. These two are the same except the first one stores the precomputed stwo trace. There is another instantiation further down which gets consumed into the trace. All of these should rely on the same instance, with methods returning stuff rather than mutating the circuit.

So:

```
let circuit = PowdrCircuit::new(self.analyzed).with_witgen_callback(...).....;
let trace = circuit.generate_stwo_trace(); // this takes a reference to the circuit and returns the trace
```

Reply:
circuit is generated only once now.